### PR TITLE
KOA-6022 Update NavBar for MD3 – View

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: Backpack/src/main/java/net/skyscanner/backpack/navbar/BpkNavBar.kt [hunk 1]
```diff
@@ -56,21 +60,6 @@ class BpkNavBar @JvmOverloads constructor(
         collapsingLayout.addView(it, params)
     }
 
-    @ColorInt
-    var expandedTitleColor: Int = context.getColor(R.color.bpkTextPrimary)
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/navbar/BpkNavBar.kt:60]: We don't use these attributes in the app

File: Backpack/src/main/java/net/skyscanner/backpack/navbar/BpkNavBar.kt [hunk 2]
```diff
@@ -84,10 +73,7 @@ class BpkNavBar @JvmOverloads constructor(
                 field = value
                 toolbar.navigationIcon = value
                 collapsingLayout.expandedTitleMarginStart = resources.getDimensionPixelSize(
-                    when (field) {
-                        null -> R.dimen.bpk_nav_bar_expanded_spacing_horizontal_small
-                        else -> R.dimen.bpk_nav_bar_expanded_spacing_horizontal_large
-                    },
+                    R.dimen.bpk_nav_bar_expanded_spacing_horizontal,
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/navbar/BpkNavBar.kt:76]: The spacing with and without the icon is now the same

File: Backpack/src/main/java/net/skyscanner/backpack/navbar/BpkNavBar.kt [hunk 3]
```diff
@@ -134,20 +118,17 @@ class BpkNavBar @JvmOverloads constructor(
...
         }
 
-        this.expandedTitleColor = expandedTitleColor
-        this.collapsedTitleColor = collapsedTextColor
         this.background = ColorDrawable(context.getColor(R.color.bpkCanvas))
         this.title = title
         this.icon = navIcon
         this.navAction = navAction
         this.menu = menu
+        this.stateListAnimator = shadowStateListAnimator()
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/navbar/BpkNavBar.kt:131]: The shadow by default is NOT 4dp, and this is how you suppose to set the preferred shadow

File: Backpack/src/main/java/net/skyscanner/backpack/navbar/internal/BpkCollapsingToolbarLayout.kt [hunk 4]
```diff
@@ -23,20 +23,29 @@ import android.graphics.drawable.ColorDrawable
...
 import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.createContextThemeWrapper
 import net.skyscanner.backpack.util.resolveThemeId
 
 internal class BpkCollapsingToolbarLayout @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0,
-) : CollapsingToolbarLayout(context, attrs, defStyleAttr) {
+) : CollapsingToolbarLayout(
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/navbar/internal/BpkCollapsingToolbarLayout.kt:33]: This is how we get the new transition animation

**Review by marianeum — APPROVED** [review]

**Pull request merged**